lib/modulesync.rb

```diff
@@ -108,7 +108,7 @@ def self.manage_file(puppet_module, filename, settings, options)
 
   def self.manage_module(puppet_module, module_files, defaults)
     puts "Syncing '#{puppet_module.given_name}'"
-    puppet_module.repository.prepare_workspace(options[:branch]) unless options[:offline]
+    puppet_module.repository.prepare_workspace(branch: options[:branch], operate_offline: options[:offline])
```

Comment:
This is changing behavior. I always used `msync update --offline` as "do not do any git actions", or `update --no-git`. It may not be entirely right, but it's a behavioral change.

Reply:
OK. Based on existing tests, I understood `offline` as a flag to operate offline, not as operate without git operations.

If I understand correctly you use `msync update --offline` to only apply templates without any modifications, could you confirm or precise the goal you want or reach using this command line?

Reply:
It is indeed. Essentially, I'd like a way to run the templates on my checkout without involving git. This is because I already prepared the whole repo, may have done some manual modifications. For example, I could have gone in and used `git add` and then use `git diff --cached` to see the exact changes. Or I did some preparation PRs to a branch. Essentially make it just a templating tool.

Reply:
So maybe a `msync templates apply` command could be a more explicit command to reach your goal?

Reply:
Probably, but that would be a major version bump. I wouldn't mind that, but I'd hate to lose the functionality now.

Reply:
OK, I revert the current behavior, ie. skipping to whole workspace preparation when running in offline mode.
I renamed the commit, as I do need to support to prepare the workspace in offline mode for other commands.

Reply:
I was just thinking: you could still make the `prepare_workspace` accept an `operate_offline` command but simply not pass it in `manage_module` and explain in a (code) comment why it's not using the parameter. Then you could still use it the other way in your other commands. Would that make sense or is it overcomplicating things?

Reply:
Checkout the updated commit, that's what I did :-)